samples/subsys/ipc/openamp/CMakeLists.txt

```diff
@@ -16,6 +16,8 @@ elseif("${BOARD}" STREQUAL "mps2_an521")
   set(BOARD_REMOTE "mps2_an521_remote")
 elseif("${BOARD}" STREQUAL "v2m_musca_b1")
   set(BOARD_REMOTE "v2m_musca_b1_ns")
+elseif("${BOARD}" STREQUAL "mimxrt1170_evk_cm7")
+  set(BOARD_REMOTE "mimxrt1170_evk_cm4")
```

Comment:
@tejlmand : AFAICT this PR just adds support for another platform to do multi-image in a sample that already does multi-image.

So it is following prior-art and should not given -1 for doing yet-another-multi-image approach.

Reply:
reading this CMakeLists.txt file in it's entirety makes this clearer.

Reply:
@SebastianBoe that's correct, this PR is another multi-image approach. I do agree that we need a generic multi image approach in Zephyr, however.

Reply:
> So it is following prior-art and should not given -1 for doing yet-another-multi-image approach.

Except for the following two things I spotted immediately.
There might be more if taking a closer look:
- The SoC code itself relies on the `REMOTE_ZEPHYR_DIR` when `DUAL_CORE_MCUX` is enabled and then does a conversion and embed the second image into the Zephyr image.
  No other SoC code does this directly, as far as I can see.
  This is a new approach
- The PR introduces `DUAL_CORE_MCUX` Kconfig setting to control when a second image should be included.
   A very generic name, yet located specific in a Kconfig.soc:
   https://github.com/zephyrproject-rtos/zephyr/blob/c10d9c4d1e39c7e533e7f48c484d7a731c26d2ee/soc/arm/nxp_imx/rt/Kconfig.soc#L656
  This is a new strategy that I would like to be discussed, especially as we have #40555 coming.
    So a new setting that might be soon deprecated is not nice.

Reply:
> embed the second image into the Zephyr image.

I could have sworn this is what the open-amp sample has always been doing. They don't do hex/bin merging.
But perhaps this has changed in the last couple of years.

Reply:
If this is not just following prior-art then the author should at least state why it is necessary to do something different.

Reply:
> > embed the second image into the Zephyr image.
> 
> I could have sworn this is what the open-amp sample has always been doing. They don't do hex/bin merging. But perhaps this has changed in the last couple of years.

This is the mess I would like to get cleaned up by having a uniform multi-image approach.
Unified in the sense it is not tied into single image Zephyr build.
Different multi-core SoC of course may have different ways they need to flash / embed multiple images.

So currently for `openamp` we have:
- lpcxpresso54114_m4 and lpcxpresso55s69_cpu0 which uses `generate_inc_file_for_gen_target()` and has a dedicated `SECOND_CORE_MCUX` setting
- mps2_an521 and v2m_musca_b1 which don't embed the second image, and don't have special setting.
   mps2_an521 even with the twist that if you build for cpu1 and not `openamp` then you always get a dummy image build for cpu0 that you cannot change. Not setting to disable and no direct Kconfig options for that image.

and now this PR which embeds the second image in another way not covered by the above methods but takes the bin file directly and embed as a section.

I prefer at this point we get a higher level handling of multi-image and thus spend time on such solution instead of yet another small variant for another SoC / board under `subsys/ipc`.